### ISSUE TITLE
testplayer failure on RPi

### ISSUE THREAD

**Issue by uzadow** [libavg/libavg#723]
From @hellbender3069:

RPi2 fails test of player
```
Running suite PlayerTestSuite
Running PlayerTest
Creating window failed: GLX is not supported. (size=[ 160, 120], bpp=24).
######## PlayerTestSuite failed. ########
FAIL: testplayer

1 of 1 test failed
Please report to uzadow@libavg.de
```
Seems it doesn't use EGL but GLX instead.

**Comment by uzadow** [comment]
I assume you're compiling libavg directly on the RPi (not crosscompiling)? I don't have a setup to test that right now, but we should be able to narrow things down. If you call `python Test.py` directly (in `src/test`), what happens?

**Comment by hellbender3069** [comment]
this is the output : 

pi@raspberrypi ~/raspberry/libavg/src/test $ python Test.py
[16-01-04 00:23:12.939][INFO][......APP] : Using libavg from: /home/pi/raspberry/libavg/src/test/libavg
[16-01-04 00:23:14.230][INFO][...CONFIG] : Display bpp: 24
[16-01-04 00:23:14.231][INFO][...CONFIG] : Requested OpenGL configuration: 
[16-01-04 00:23:14.231][INFO][...CONFIG] :   OpenGL flavor: Mobile (ES)
[16-01-04 00:23:14.232][INFO][...CONFIG] :   Pixel buffers: true
[16-01-04 00:23:14.232][INFO][...CONFIG] :   Power of 2 textures: false
[16-01-04 00:23:14.233][INFO][...CONFIG] :   No multisampling
[16-01-04 00:23:14.233][INFO][...CONFIG] :   Shader usage: auto
[16-01-04 00:23:14.233][INFO][...CONFIG] :   Debug context: true
Traceback (most recent call last):
  File "Test.py", line 146, in <module>
    app.run()
  File "/home/pi/raspberry/libavg/src/test/testapp.py", line 73, in run
    self.__setupTestApp()
  File "/home/pi/raspberry/libavg/src/test/testapp.py", line 110, in __setupTestApp
    self.__dumpConfig()
  File "/home/pi/raspberry/libavg/src/test/testapp.py", line 187, in __dumpConfig
    player.play()
libavg.avg.Exception: Creating window failed: GLX is not supported. (size=[      160,      120], bpp=24).
pi@raspberrypi ~/raspberry/libavg/src/test $

**Comment by uzadow** [comment]
Hm - this works on my machine (I tested it yesterday). So let's find out what the differences are. I used cross-compilation and started the tests using a shell in a running X server. Also - can you post the results of the configure step?

Alternatively, are you up to debugging into the display setup?

**Comment by hellbender3069** [comment]
I have attached the output of 

./configure --enable-rpi --enable-egl

I will check my X drivers just to be shure, but it is a stock rasbian, so they should be ok. I have tested with the hello_pi tests and all of them work.

I also start the tests from within a shell inside X.

Which IDE do you use to debug? I can handle that if I know a good IDE ( sorry i program in nano :-) on linux )
[configure_output.txt](https://github.com/libavg/libavg/files/77438/configure_output.txt)

**Comment by uzadow** [comment]
Thanks. The configure output looks ok at first sight. I'll compare it to mine when I get home later tonight.

There's no good IDE, it's either gdb or printing to the console. If/when we move to CMake, we can use CLion as IDE, but not now.

**Comment by hellbender3069** [comment]
Wait before you check the config. Maybe the problem is simpler than that. 

When I checked my X.org.log it said it was using the AllWinner A10/A20 FB driver.....
Thats not correct :-S.... oh wait..maybe this is the image of a hummingboard which i have laying around.

I am going to flash the SDCard with a new RPI2 Rasbian, instead of the current. That could be the problem since the Hummingboard has another videochip inside.

My fault. I will let you know as soon as i got everything back up and running

**Comment by hellbender3069** [comment]
Nope I am sorry, with raspbian jessy still the same problem. It seems to do with the video driver in X but I can't figure it out yet. It reverts back to software rendering instead of loading the EGL driver or something like that.

Are you running Wheezy or Jessy?

**Comment by hellbender3069** [comment]
Ok I am not getting any further. 

I tried Wheezy, but you can't compile libAVG on Wheezy anymore because there is no libSDL2.0.2 available. It has 1.2 and in the backports repo I can find 2.0.0 which is rejected by the configure scipt.

Ok back to Jessie.
1) Installed a clean image of Jessie downloaded directly from raspberrypi.org.
2) apt-get update and upgrade
3) create a 2 gig swap
4) download the prerequisits and git
5) download libavg from git ( my repo but there is no change to any code accept libavcoded calls )
6) export the archflags for RPi2
7) ./bootstrap and ./configure --enable-rpi --enable-egl
8) build, no errors during build
9) in src/test Python Test.py -- error no GLX display
10) found in /var/log/X.org.0.log found its missing software rendering library
11) installed libDRI and libDRI2 ( which seem to be missing in the image )
12 ) reboot and check X.org.0.log, now software rendering is enabled.
13 ) in src/test run python Test.py -- error no EGL display
14) ran /opt/vc/src/hello_pi/hello_teacup/hello_teacup ( which uses EGL to render ) works just fine
15) ran src/player/testplayer ( no EGL ) found this error somewhere in graphics, so that is where it tries to create an EGL display but fails.

do you have a binary of your build I could try to see if it is the build or if it is the raspbian image??

thanks,
Michel

**Comment by uzadow** [comment]
Sorry - there was a major omission in the install docs. libSDL needs to be installed from source for this to work. I've updated the wiki appropriately. Also, update libavg again, because I've updated the libSDL used to 2.0.4 and that caused some incompatibilities (before, I used git head).

Again, sorry. My fault. If you run into any more issues, don't hesitate to ask again.

Best,

  Uli

**Comment by hellbender3069** [comment]
It might be a good idea to mention in the Wiki that you are running this on raspbian wheezy and not on raspbian jessie, since jessie is the newest version and people will automaticly download that one from the RPi website.

I will use wheezy and update the test to use the source of libSDL.

Thanks and I will keep you posted

**Comment by hellbender3069** [comment]
Ok the problem with not being able to create an EGL display seems to be an issue with the jessie image. I couldn't get it to work so I reverted to wheezy. Build everything as mentioned on the libAVG cross-compile blog but compiled native.

Ran the tests and they ran this time but ended with three failures. Do you have any idea if these are critical? Resultimages are in the attachment.

======================================================================
FAIL: testCurve (VectorTest.VectorTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/pi/raspberry/libavg/src/test/VectorTest.py", line 307, in testCurve
    lambda: self.compareImage("testCurve4"),
  File "/home/pi/raspberry/libavg/src/test/testcase.py", line 141, in start
    player.play()
  File "/home/pi/raspberry/libavg/src/test/testcase.py", line 271, in __nextAction
    action()
  File "/home/pi/raspberry/libavg/src/test/VectorTest.py", line 301, in <lambda>
    (lambda: self.compareImage("testCurve1"),
  File "/home/pi/raspberry/libavg/src/test/testcase.py", line 152, in compareImage
    self.compareBitmapToFile(bmp, fileName)
  File "/home/pi/raspberry/libavg/src/test/testcase.py", line 179, in compareBitmapToFile
    self.fail(msg)
AssertionError:   testCurve1: Difference image has avg=0.20, std dev=7.12

======================================================================
FAIL: testMesh (VectorTest.VectorTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/pi/raspberry/libavg/src/test/VectorTest.py", line 752, in testMesh
    lambda: handlerTester.assertState(
  File "/home/pi/raspberry/libavg/src/test/testcase.py", line 141, in start
    player.play()
  File "/home/pi/raspberry/libavg/src/test/testcase.py", line 271, in __nextAction
    action()
  File "/home/pi/raspberry/libavg/src/test/VectorTest.py", line 732, in <lambda>
    lambda: self.compareImage("testMesh4"),
  File "/home/pi/raspberry/libavg/src/test/testcase.py", line 152, in compareImage
    self.compareBitmapToFile(bmp, fileName)
  File "/home/pi/raspberry/libavg/src/test/testcase.py", line 179, in compareBitmapToFile
    self.fail(msg)
AssertionError:   testMesh4: Difference image has avg=0.27, std dev=6.94

=====================================================================
FAIL: testCanvasMipmap (OffscreenTest.OffscreenTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/pi/raspberry/libavg/src/test/OffscreenTest.py", line 385, in testCanvasMipmap
    self.start(False, (lambda: self.compareImage("testCanvasMipmap"),))
  File "/home/pi/raspberry/libavg/src/test/testcase.py", line 141, in start
    player.play()
  File "/home/pi/raspberry/libavg/src/test/testcase.py", line 271, in __nextAction
    action()
  File "/home/pi/raspberry/libavg/src/test/OffscreenTest.py", line 385, in <lambda>
    self.start(False, (lambda: self.compareImage("testCanvasMipmap"),))
  File "/home/pi/raspberry/libavg/src/test/testcase.py", line 152, in compareImage
    self.compareBitmapToFile(bmp, fileName)
  File "/home/pi/raspberry/libavg/src/test/testcase.py", line 179, in compareBitmapToFile
    self.fail(msg)
AssertionError:   testCanvasMipmap: Difference image has avg=1.20, std dev=11.76

----------------------------------------------------------------------
Ran 275 tests in 115.047s
[resultimages.tar.gz](https://github.com/libavg/libavg/files/83907/resultimages.tar.gz)

**Comment by uzadow** [comment]
The three test failures you're getting are minor rendering inaccuracies (that's this bug: #627).

Also, I've reproduced the failing Jaunty build and opened a bug for that: #729.

**Comment by hellbender3069** [comment]
Great now i can continue my work on openmax. lets see if we can get more progress then last time.

Jessie will have to wait until they fixed the EGL driver problem in X.

**Comment by uzadow** [comment]
Cool.

Which "EGL driver problem in X" are you referring to? Right now (on my jaunty setup), tests run & look mostly ok on screen, but the screenshots are broken.

**Comment by hellbender3069** [comment]
I have tested with Jessie downloaded from the RPi website itself. It seems to be having trouble with the EGL drivers for X. The mesa drivers install libraries with the same name but other functions. Which results in the error messages I posted earlier. But I used the libSDL packages in the repository and that could be a problem. I will try and build libSDL from scratch as you do it, since that forces the system to use openGL ES.

Jaunty is a ubuntu flavor if i am right??
